frontend/src/metabase/lib/time.ts

```diff
@@ -197,3 +197,45 @@ export function parseTimestamp(
   }
   return isLocal ? m.local() : m;
 }
+
+const isTimezoneInOffsetFormat = (timezone: string) => {
+  const offsetPattern = /^[+-]\d{2}:\d{2}$/;
+  return offsetPattern.test(timezone);
+};
+
+const getTimezoneWithOffset = (offsetString: string) => {
+  const offsetSign = offsetString[0];
+  const offsetHours = parseInt(offsetString.substring(1, 3));
+  const offsetMinutes = parseInt(offsetString.substring(4, 6));
+
+  const totalOffsetMinutes =
+    (offsetHours * 60 + offsetMinutes) * (offsetSign === "+" ? 1 : -1);
+
+  const matchingZones = moment.tz.names().filter(zone => {
+    const tzOffset = moment.tz(zone).utcOffset();
+    return tzOffset === totalOffsetMinutes;
+  });
+
+  const etcZone = matchingZones.find(zone => zone.startsWith("Etc/GMT"));
```

Comment:
might be nice to add a small comment here on why we prioritize Etc/GMT here instead of just using the first matching timezone

Reply:
I realized that this logic was not correct for arbitrary offsets like +9:30. Even if there is an IANA time zone for such value, it can be a DST timezone. So it would be more correct to just apply the offset manually